src/common.h

```diff
@@ -369,6 +369,24 @@ char *ly_strnchr(const char *s, int c, size_t len);
  */
 int ly_strncmp(const char *refstr, const char *str, size_t str_len);
 
+/**
+ * @brief Get all possible prefixes from an arbitrary YANG string by iteratively returning specific substrings.
+ *
+ * The function looks for possible prefix at the beginning of the @p str_begin and returns it.
+ * If there is some string at the @p str_begin and then follows the prefix, then the string is returned.
+ * If @p str_begin does not contain any prefix, then the entire @p str_begin is returned.
+ * The @p next can be used as the @p str_begin in the next iteration.
+ *
+ * @param[in] str_begin Begin of the input string.
+ * @param[in] str_end Length of the @p str_begin. If set to NULL then the @p str_begin must be NULL-terminated string.
+ * @param[out] is_prefix Type of substring found. Set to 1 for prefix, otherwise 0.
+ * @param[out] str_next Rest of the string after the found substring.
+ * If the @p is_prefix is set to 1 then the colon character is skipped.
+ * If no string remains, it is set to NULL.
+ * @return Number of bytes (length) of the found substring starting at @p str_begin.
+ */
+size_t ly_string_next(const char *str_begin, const char *str_end, int *is_prefix, const char **str_next);
```

Comment:
I have not noticed before, why is `is_prefix` `int`? It should definitely be a `ly_bool`.

Reply:
ok